content/en/docs/languages/net/instrumentation.md

```diff
@@ -22,361 +36,581 @@ are covered here as well as the `System.Diagnostics` API.
-(Semantic Conventions) that aren't yet classed as stable.
 
-Next, configure it in your ASP.NET Core startup routine where you have access to
-an `IServiceCollection`.
+### Initialize the SDK
+
+{{% alert title="Note" color="info" %}} If you’re instrumenting a library,  you dont need to initialize the sdk. {{% /alert %}}
+
+It is important to configure an instance of the OpenTelemetry SDK as early as possible in your application.
+
+To initialize the OpenTelemetry SDK for an ASP.NET Core app like in the case of the example app, 
+replace the content of the program.cs file with the following code: 
 
 ```csharp
-using OpenTelemetry;
+using OpenTelemetry.Logs;
+using OpenTelemetry.Metrics;
 using OpenTelemetry.Resources;
 using OpenTelemetry.Trace;
 
-// Define some important constants and the activity source.
-// These can come from a config file, constants file, etc.
-var serviceName = "MyCompany.MyProduct.MyService";
+// Ideally, you will want this name to come from a config file, constants file, etc.
+var serviceName = "Dice.*";
 var serviceVersion = "1.0.0";
 
 var builder = WebApplication.CreateBuilder(args);
 
-// Configure important OpenTelemetry settings, the console exporter
 builder.Services.AddOpenTelemetry()
-  .WithTracing(b =>
-  {
-      b
-      .AddSource(serviceName)
-      .ConfigureResource(resource =>
-          resource.AddService(
-            serviceName: serviceName,
-            serviceVersion: serviceVersion))
-      .AddAspNetCoreInstrumentation()
-      .AddConsoleExporter();
-  });
+    .ConfigureResource(resource => resource.AddService(
+        serviceName: serviceName, 
+        serviceVersion: serviceVersion))
+    .WithTracing(tracing => tracing
+        .AddSource(serviceName)
+        .AddConsoleExporter())
+    .WithMetrics(metrics => metrics
+        .AddMeter(serviceName)
+        .AddConsoleExporter());
+
+builder.Logging.AddOpenTelemetry(options => options
+    .SetResourceBuilder(ResourceBuilder.CreateDefault().AddService(
+        serviceName: serviceName,
+        serviceVersion: serviceVersion))
+    .AddConsoleExporter());
+
+builder.Services.AddControllers();
+
+var app = builder.Build();
+
+app.MapControllers();
+
+app.Run();
+
+```
+
+If initializing the OpenTelemetry SDK for a console app,
+add the following code at the beginning of your program, during any important startup operations.
+
+```csharp
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+
+//...
+
+var serviceName = "MyServiceName";
+var serviceVersion = "1.0.0";
+
+var tracerProvider = Sdk.CreateTracerProviderBuilder()
+    .AddSource(serviceName)
+    .ConfigureResource(resource =>
+        resource.AddService(
+          serviceName: serviceName,
+          serviceVersion: serviceVersion))
+    .AddConsoleExporter()
+    .Build();
+
+var meterProvider = Sdk.CreateMeterProviderBuilder()
+    .AddMeter(serviceName)
+    .AddConsoleExporter()
+    .Build();
+
+//...
```

Comment:
logs omitted intentionally?

Reply:
I missed it. 

Thanks for pointing it out 🙂